plugins/modules/rabbitmq_plugin.py

```diff
@@ -139,6 +152,7 @@ def main():
     arg_spec = dict(
         names=dict(required=True, aliases=['name']),
         new_only=dict(default='no', type='bool'),
+        node=dict(default='rabbit'),
```

Comment:
I was just wondering, by specifying `default='rabbit'` here would this always populate `self.module.params['node']` with `rabbit` regardless of what's in the playbook?

I was looking through the rabbitmq.server code on github, but, it seems quite time consuming to go back through all the versions looking for when `-n` was introduced to rabbitmqctl/rabbitmq-plugins...

I was wondering, can we guarantee backward compatibility by changing line 155 to:

`node=dict(type='str')`

At least then, the user can chose not to put `node` in their playbook at it will work as before.  What do you think?

Reply:
The option is here at least since rabbitmq:3.4.4 ( 11 February 2015 )

```
$ docker run --rm -ti --entrypoint /bin/bash rabbitmq:3.4.4
root@cfd69b2f4d2c:/# rabbitmq-plugins
Error: could not recognise command
Usage:
rabbitmq-plugins [-n <node>] <command> [<command options>] 

Commands:
    list [-v] [-m] [-E] [-e] [<pattern>]
    enable [--offline] [--online] <plugin> ...
    disable [--offline] [--online] <plugin> ...
    set [--offline] [--online] <plugin> ...


root@cfd69b2f4d2c:/#
```

Reply:
Thanks for checking that.  It appears that 3.4.4 is the earliest docker container.

> I personally prefer to have all modules working the same that having some specificities depending on the module

The only risk with the proposed PR is that it will specify `-n` on the command line all the time as default specifies `rabbit`.  In it self, that is fine but _if_ someone is using an older than 3.4.4. version which does not support `-n` and they use the latest community.rabbitmq version a rabbitmq_plugin playbook will fail.  I was just trying to avoid this potential backwards compatibility issue... although it maybe quite a small risk?

If we proceed as is, perhaps we consider adjusting the changelog fragment to include a message that this will be a breaking change if the users `rabbitmq-plugins` does not support the `-n` flag.  How does that sound?

Reply:
IMO all module should not force options if not mandatory but for an obscure reason `-n` seems forced in many rabbitmq module and is the only reason why I add it.
So it is your call here, but it will be more consistent to apply this choice to all modules in a future PR.

Off-topic : For an example, https://github.com/ansible-collections/community.rabbitmq/pull/35 broke my ansible role because the choice was made to force `--online` when this option is not mandatory and `rabbitmq-plugins` is able to do the choice itself.....

Reply:
@Im0 / @Andersson007 : so, force or not force `-n` ?

Reply:
@cognifloyd @Im0 @odyssey4me what do you think?

@rockandska I'm unfortunately not an expert in the context, so i can provide only general feedback.
I can say that if we're gonna introduce any breaking changes, it would be nice:
1) to announce this in advance, i.e. that it's gonna happen in the next major release or so
2) release the major version in some time after the announcement

Reply:
@cognifloyd @Im0 @odyssey4me PTAL ^ Hope you folks have great holidays:)